core/signer/signer.go

```diff
@@ -115,20 +114,59 @@ func (s *Signer) ProcessMutation(ctx context.Context, txn transaction.Txn, index
 		return fmt.Errorf("QueueLeaf err: %v", err)
 	}
 
-	mutationObj := new(tpb.SignedKV)
-	if err := proto.Unmarshal(mutation, mutationObj); err != nil {
-		return err
+	log.Printf("Sequenced %x", index)
+	return nil
+}
+
+// processMutations reads mutations from the database and adds them to the tree.
+// processMutations returns the maximum mutation sequence number processed.
+func (s *Signer) processMutations(ctx context.Context, txn transaction.Txn) (uint64, error) {
+	startSequence := uint64(0)
+	smh := new(ctmap.SignedMapHead)
+	if _, _, err := s.sths.Latest(ctx, smh); err == nil {
+		startSequence = smh.GetMapHead().MaxSequenceNumber + 1
```

Comment:
While this works for sequence numbers, it also works for timestamp. As long as we provide a starting point that is greater than `MaxSequenceNumber` but not greater than the next sequence number after `MaxSequenceNumber` then the SQL query will cover it. Note that the SQL query has `Sequence >= ?` as a condition. This with the `+1` is basically a ( ] interval.

The SQL query is already collecting a ( ] interval.